File: Lab_04/lang-tests/test-4/alcance.java
public class alcance
{
    static int a = 5;

    public static int g()
    {
        return a;
    }

    public static void main(String []args)
    {
        int a = 0;
        int b = g();
        System.out.println("b vale: " + b);
    }
}

File: Lab_04/lang-tests/test-4/alcance.scala
object alcance
{
    def main(args: Array[String])
    {
        var a : Int = g()
        println(a)
    }
    var x = 2
    def f(): Int ={
        return x
    }
    
    def g(): Int ={
        var x = 3
        return f()
    }
}